lib/APIPlugin.js

```diff
@@ -39,12 +39,16 @@ class APIPlugin {
 			compilation.dependencyFactories.set(ConstDependency, new NullFactory());
 			compilation.dependencyTemplates.set(ConstDependency, new ConstDependency.Template());
 
-			normalModuleFactory.plugin(["parser javascript/auto", "parser javascript/dynamic", "parser javascript/esm"], parser => {
+			const handler = parser => {
 				Object.keys(REPLACEMENTS).forEach(key => {
 					parser.plugin(`expression ${key}`, NO_WEBPACK_REQUIRE[key] ? ParserHelpers.toConstantDependency(REPLACEMENTS[key]) : ParserHelpers.toConstantDependencyWithWebpackRequire(REPLACEMENTS[key]));
 					parser.plugin(`evaluate typeof ${key}`, ParserHelpers.evaluateToString(REPLACEMENT_TYPES[key]));
 				});
-			});
+			};
+
+			normalModuleFactory.hooks.parser.for("javascript/auto").tap("APIPlugin", handler);
```

Comment:
You can also do this instead: 

``` js
normalModuleFactory.hooks.parser.tap("javascript/auto", "APIPlugin", handler);
```

Reply:
Do you want me to do the change? I think `.for(x).tap(` is cleaner.

Reply:
It's ok. You don't have to change it.